types.go

```diff
@@ -55,6 +55,11 @@ type (
 		// Checksum ensures the integrity of packages
 		// refereced by URL. Ignored for literals.
 		Checksum Checksum `json:"checksum"`
+	}
+
+	PackageRef struct {
+		PackageName      string
+		PackageNamespace string
```

Comment:
There are no versions in packages. If a package is mutated, the reference now points at a changed package, and the function has changed.